From 76a3654537472ad7be7f740c144d666364c11e3f Mon Sep 17 00:00:00 2001
From: Hoyho Luo <luohaihao@gmail.com>
Date: Wed, 5 Jan 2022 08:57:07 +0800
Subject: [PATCH 2/4] fix: parse invalid lyric timestamp failed (#938)

Time string from lyric_line is invalid in some case
For example, '99:00.00' which does not match format '%M:%S.%f'

fix #936

Signed-off-by: hoyho <luohaihao@gmail.com>
---
 NEMbox/ui.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/NEMbox/ui.py b/NEMbox/ui.py
index 650bc98..8ef7376 100644
--- a/NEMbox/ui.py
+++ b/NEMbox/ui.py
@@ -184,7 +184,7 @@ class Ui(object):
 
     def update_lyrics(self, now_playing, lyrics, tlyrics):
 
-        timestap_regex = r"\d\d:\d\d\.[0-9]*"
+        timestap_regex = r"[0-5][0-9]:[0-5][0-9]\.[0-9]*"
 
         def get_timestap(lyric_line):
             match_ret = re.match(r"\[(" + timestap_regex + r")\]", lyric_line)
-- 
2.34.1

